### PR TITLE
Fix sns permissions

### PR DESCRIPTION
**Pull request by leondutoit**
Also ensure API can run without mq config.

### CODE REVIEW

File: tsdfileapi/api.py [hunk 1]
```diff
@@ -531,6 +531,9 @@ def write_file(self, filemode, filename, filebody, tenant):
                 os.chmod(self.path_part, _RW_RW___)
             self.new_paths.append(self.path_part)
             if self.backend == 'sns':
+                if not os.path.lexists(tsd_hidden_folder):
```
Comment (haatveit) [inline tsdfileapi/api.py:534]: Will the following lines here ever run? A os.path.lexists() check runs as part of the sns_dir() call that assigns tsd_hidden_folder's value.

File: tsdfileapi/utils.py [hunk 2]
```diff
@@ -81,20 +81,19 @@ def sns_dir(base_pattern, tenant, uri, tenant_string_pattern, test=False):
...
+        assert _IS_REALISTIC_PGP_KEY_FINGERPRINT.match(keyid), f'invalid PGP fingerprint: {keyid}'
         folder = base_pattern.replace(tenant_string_pattern, tenant).replace('KEYID', keyid).replace('FORMID', formid)
         _path = os.path.normpath(folder)
         if test:
             return _path
         if not os.path.lexists(_path):
             logging.info('Creating %s', _path)
             os.makedirs(_path)
-            subprocess.call('chmod', '775', _path)
+            subprocess.call(['chmod', '2770', _path])
```
Comment (haatveit) [inline tsdfileapi/utils.py:93]: Wouldn't this have caused a TypeError on every new invocation? Now I get it.

Comment (leondutoit) [reply tsdfileapi/utils.py:93]: it did yes

File: tsdfileapi/api.py (hunk 1 above)
Comment (leondutoit) [reply tsdfileapi/api.py:534]: ack.

**Pull request merged**